tests/test_web/test_api_v1.py

```diff
@@ -117,57 +117,96 @@ def test_get_builds_invalid_state(app, client, db):
+                'add_arches': ['s390x'],
+            },
+            'add',
             '"bundles" should be a non-empty array of strings',
         ),
-        (['some:thing'], 32, 'binary:image', ['s390x'], '"from_index" must be a string'),
         (
-            ['something'],
-            'pull_spec',
-            'binary_image',
-            [1, 2, 3],
+            {
+                'bundles': ['some:thing'],
+                'from_index': 32,
+                'binary_image': 'binary:image',
+                'add_arches': ['s390x'],
+            },
+            'add',
+            '"from_index" must be a string',
+        ),
+        (
+            {
+                'bundles': ['something'],
+                'from_index': 'pull_spec',
+                'binary_image': 'binary_image',
+                'add_arches': [1, 2, 3],
+            },
+            'add',
             'Architectures should be specified as a non-empty array of strings',
         ),
+        (
+            {
+                'operators': [],
+                'from_index': 'pull:spec',
+                'binary_image': 'binary:image',
+                'add_arches': ['s390x'],
+            },
+            'rm',
+            '"operators" should be a non-empty array of strings',
+        ),
     ),
 )
-def test_add_bundle_invalid_params_format(
-    bundles, from_index, binary_image, add_arches, error_msg, db, auth_env, client
-):
-    data = {
-        'bundles': bundles,
-        'from_index': from_index,
-        'binary_image': binary_image,
-        'add_arches': add_arches,
-    }
-
-    rv = client.post('/api/v1/builds/add', json=data, environ_base=auth_env)
+def test_add_or_rm_invalid_params_format(data, request_type, error_msg, db, auth_env, client):
+    rv = client.post(f'/api/v1/builds/{request_type}', json=data, environ_base=auth_env)
     assert rv.status_code == 400
     assert error_msg == rv.json['error']
 
 
-def test_add_bundle_missing_required_param(db, auth_env, client):
-    data = {
-        'from_index': 'from_index',
-        'binary_image': 'binary:image',
-        'add_arches': ['add_arches'],
-    }
-
-    rv = client.post('/api/v1/builds/add', json=data, environ_base=auth_env)
+@pytest.mark.parametrize(
+    'data, request_type, error_msg',
+    (
+        (
+            {'from_index': 'pull:spec', 'binary_image': 'binary:image', 'add_arches': ['s390x1']},
+            'add',
+            'Missing required parameter(s): bundles',
+        ),
+        (
+            {'bundles': ['some:thing'], 'from_index': 'pull:spec', 'add_arches': ['s390x1']},
+            'add',
+            'Missing required parameter(s): binary_image',
+        ),
+        (
+            {'from_index': 'pull:spec', 'binary_image': 'binary:image', 'add_arches': ['s390x1']},
+            'rm',
+            'Missing required parameter(s): operators',
+        ),
+    ),
+)
+def test_add_or_rm_missing_required_param(data, request_type, error_msg, db, auth_env, client):
```

Comment:
Could you make these separate tests (one for add and one for rm)? By keeping them together, you only save 3 lines of code and it's harder to read.